code/clue.js

```diff
@@ -1,59 +1,408 @@
+const mystery = {
+  killer: undefined,
+  weapon: undefined,
+  room: undefined 
+};
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// FUNCTIONS
 
-// This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  mystery.killer = randomSelector(suspects);
+
+  const killerImage = document.getElementById("killerImage");
+  const favouriteWeapon = document.getElementById("favouriteWeapon");
+  const killerAge = document.getElementById("killerAge");
+  const killerName = document.getElementById("killerName");
+
+  killerImage.style.display = "block";
+  favouriteWeapon.style.display = "block";
+  killerAge.style.display = "block";
+  killerName.style.display = "block";
+  killerCard.style.background = mystery.killer.color;
+  document.getElementById("loadingAnimation1").style.display = "none";
+
+  killerName.innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  killerImage.src = `${mystery.killer.image}`;
+  killerAge.innerHTML = `Age: ${mystery.killer.age}`;
+  favouriteWeapon.innerHTML = `Favourite Weapon: ${mystery.killer.favouriteWeapon}`;
+};
+
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons);
+
+  const weaponImage = document.getElementById("weaponImage");
+  const weaponName = document.getElementById("weaponName");
+  const weaponWeight = document.getElementById("weaponWeight");
+
+  weaponName.style.display = "block";
+  weaponWeight.style.display = "block";
+  weaponImage.style.display =  "block";
+  document.getElementById("loadingAnimation2").style.display = "none";
+
+  weaponCard.style.background = mystery.weapon.color;
+  weaponName.innerHTML = `${mystery.weapon.name}`;
+  weaponWeight.innerHTML = `${mystery.weapon.weight}`;
+  weaponImage.src = `${mystery.weapon.image}`;
+};
+
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms);
+
+  const roomCard = document.getElementById("roomCard");
+  const roomName = document.getElementById("roomName");
+  const roomImage = document.getElementById("roomImage");
+
+  roomName.style.display = "block";
+  roomImage.style.display = "block";
+  document.getElementById("loadingAnimation3").style.display = "none";
+
+  roomCard.style.background = mystery.room.color;
+  roomName.innerHTML = `${mystery.room.name}`;
+  roomImage.src = `${mystery.room.image}`;
+};
+
+const startAnimation1 = () => {
+  const killerImage = document.getElementById("killerImage");
+  const favouriteWeapon = document.getElementById("favouriteWeapon");
+  const killerAge = document.getElementById("killerAge");
+  const killerName = document.getElementById("killerName");
+
+  killerImage.style.display = "none";
+  favouriteWeapon.style.display = "none";
+  killerAge.style.display = "none";
+  killerName.style.display = "none";
```

Comment:
This section contains all the killer data, maybe can a `<section>` be created in HTML to contain all this data and then change that section to `display: none;` instead of doing it for every line separately?
Same for lines 369-371 and 381-382

Also, this initial `display: none;` styling, can it be set like that in the CSS since the start (instead of here in the JS file)
and then changed to `display: block; `when the pickKiller, pickWeapon and pickRoom functions are called?

Reply:
Interesting, so making a container that hides, rather than hiding each element?